### PR TITLE
Enhancement/#314 Add a tooltip to the ButtonV2 component in the 'icon' variant

### PR DESCRIPTION
**Pull request by nsstarr**
## What's done
- Added tooltip to the ButtonV2 component with the title prop "Click here to check your wallet balance"
- Ensured that only the button with the 'icon' variant has a tooltip according to Design System v2.

## How to test

Run the storybook script in .src and interact with the ButtonV2 Icon component.

## Tasks

- [ ] Have you written tests for new code (if applicable)? I will add tests if they are required
- [ ] Have you tested the changes on all the platforms?
  - [ ] iOS
  - [ ] Android
  - [x] Web
- [x] Have you added stories to demonstrate the new functionality (if there is any)? 

## Demo (screenshots, recordings, etc.)

https://github.com/roll-network/tryrolljs/assets/98424384/2f6abf69-35d8-4514-9d43-05c3a162c932

### CODE REVIEW

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 1]
```diff
@@ -118,71 +119,141 @@ const BaseButton = ({
...
-            {iconVariant && variant !== 'text' && (
-              <View style={[styles.iconContainer]}>
-                <Icon
-                  variant={iconVariant}
-                  width={iconBasedOnSize[size]}
-                  height={iconBasedOnSize[size]}
-                  color={iconColor || stylesBasedOnState.textColor}
-                />
-              </View>
+      {variant === 'icon' ? (
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/index.tsx:122]: This approach includes a huge code duplication. Instead, you can do something like this:
```suggestion
  // Inside `ButtonV2` scope
  const button = (
    <Pressable
            testID={testID}
            isDisabled={isLoading || isDisabled}
            onHoverIn={() => setIsHover(true)}
            onHoverOut={() => setIsHover(false)}
            onPressIn={() => setIsActive(true)}
            onPressOut={() => setIsActive(false)}
            onPress={onPress}
            isHovered={isHover}
            // Styles
            position="relative"
            alignItems="center"
            justifyContent="center"
            paddingX={`${paddingX}px`}
            paddingY={`${paddingY}px`}
            borderRadius={borderRadius}
            borderColor={stylesBasedOnState.borderColor}
            borderWidth={rest.borderWidth}
            backgroundColor={
              iconBackgroundColor || stylesBasedOnState.backgroundColor
            }
            flexDirection="row"
            _hover={{
              borderColor: hover.borderColor,
              borderWidth: hover.borderWidth,
            }}
            _focus={{
              backgroundColor: iconBackgroundColor || active.backgroundColor,
            }}
            _pressed={{
              borderColor: active.borderColor,
              borderWidth: active.borderWidth,
              backgroundColor: iconBackgroundColor || active.backgroundColor,
            }}
          >
            {isLoading ? (
              <Spinner size={lineHeight} color={stylesBasedOnState.textColor} />
            ) : (
              <>
                {iconVariant && (
                  <View style={[styles.iconContainer]}>
                    <Icon
                      variant={iconVariant}
                      width={iconBasedOnSize[size]}
                      height={iconBasedOnSize[size]}
                      color={iconColor || stylesBasedOnState.textColor}
                    />
                  </View>
                )}
                {!iconVariant && icon && (
                  <View style={[styles.iconContainer]}>{icon}</View>
                )}
                {variant !== 'icon' && (
                  <TypographyV2
                    variant={fontBasedOnVariantAndSize}
                    selectable={false}
                    color={stylesBasedOnState.textColor}
                    style={[styles.title]}
                  >
                    {title}
                  </TypographyV2>
                )}
              </>
            )}
          </Pressable>
    )
    
    return variant === 'icon' ? <TooltipV2 title={title}>{button}</TooltipV2> : button
```

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 2]
```diff
@@ -118,71 +119,141 @@ const BaseButton = ({
...
+        <TooltipV2 title='Click here to check your wallet balance'>
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/index.tsx:123]: The tooltip title should be configurable (not static)

```suggestion
        <TooltipV2 title={title}>
```

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 3]
```diff
@@ -302,6 +373,7 @@ const useVariantsProps = (variant: Variant): VariantProps => {
         },
       }
     }
+    //Button Icon Variant
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/index.tsx:376]: The comment seems redundant here

```suggestion
```

File: packages/design-system/src/atoms/buttonV2/types.ts [hunk 4]
```diff
@@ -39,6 +40,7 @@ export type StateVariantProps = {
   borderColor: string
   backgroundColor: string
   textColor: string
+  setIsTooltipVisible?: boolean
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/types.ts:43]: Looks like the prop is unused

```suggestion
```

File: packages/design-system/src/atoms/tooltipV2/index.tsx [hunk 5]
```diff
@@ -5,6 +5,7 @@ import { TypographyV2 } from '../typographyV2'
 export interface TooltipV2Props
   extends Omit<TooltipProps, 'title' | 'variant'> {
   title: string
+  visible?: boolean
```
Comment (teimurjan) [inline packages/design-system/src/atoms/tooltipV2/index.tsx:8]: What's the reason for this change?

**Review by teimurjan — CHANGES_REQUESTED** [review]
Great start! :) There are some things you need to fix before the merge:

1. Address all the comments.
2. Test the change on native (`yarn start:native`) & provide screen recordings.
3. Add a changeset file.

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 6]
```diff
@@ -116,75 +117,85 @@ const BaseButton = ({
...
-                style={[styles.title]}
-              >
-                {title}
-              </TypographyV2>
-            )}
-          </>
-        )}
-      </Pressable>
+    
+    <View style={[container.alignCenter, style]}>
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/index.tsx:189]: It looks like prettier hasn't run here. Can you run `Format` in VSCode, please?

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 7]
```diff
@@ -116,75 +117,85 @@ const BaseButton = ({
...
-                {title}
-              </TypographyV2>
-            )}
-          </>
-        )}
-      </Pressable>
+    
+    <View style={[container.alignCenter, style]}> 
+    {variant === 'icon' ?
+    <TooltipV2 title={title}>{button}</TooltipV2>
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/index.tsx:191]: I think `title` is not the prop we have to pass here. `title` relates to the Button's text. We have to allow add any text to the Tooltip. You've already added `tooltip` prop in the `types.ts`. I think you need to use it.

```suggestion
    <TooltipV2 title={tooltip}>{button}</TooltipV2>
```

File: packages/design-system/src/atoms/buttonV2/index.tsx (hunk 6 above)
Comment (nsstarr) [reply packages/design-system/src/atoms/buttonV2/index.tsx:189]: Thank you for pointing it out. Will do & commit the formatted version.

File: packages/design-system/src/atoms/tooltipV2/index.tsx (hunk 5 above)
Comment (nsstarr) [reply packages/design-system/src/atoms/tooltipV2/index.tsx]: That change is now removed as it's redundant. It was added when I was trying out another solution.

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 8]
```diff
@@ -116,73 +118,80 @@ const BaseButton = ({
...
-                color={stylesBasedOnState.textColor}
-                style={[styles.title]}
-              >
-                {title}
-              </TypographyV2>
-            )}
-          </>
-        )}
-      </Pressable>
+      {variant === 'icon' ? (
```
[the remark this answers is not in the document]
Comment (nsstarr) [reply packages/design-system/src/atoms/buttonV2/index.tsx:190]: Thanks for suggestion! I'll test and push the changes

[the remark this answers is not in the document]
Comment (nsstarr) [reply packages/design-system/src/atoms/buttonV2/index.tsx]: I've implemented the Tooltip in buttonv2 component with a Conditional Wrapper. Please see the most recent commit.

File: packages/design-system/src/atoms/tooltipV2/index.tsx [hunk 9]
```diff
@@ -4,7 +4,7 @@ import { TypographyV2 } from '../typographyV2'
 
 export interface TooltipV2Props
   extends Omit<TooltipProps, 'title' | 'variant'> {
-  title: string
+  title: string | undefined
```
Comment (teimurjan) [inline packages/design-system/src/atoms/tooltipV2/index.tsx:7]: That's not correct. Tooltips should always have a title, otherwise, it'll render an empty container. Instead, you have to check in the ButtonV2 component & show the tooltip only if the `tooltip` prop is defined.

**Review by teimurjan — CHANGES_REQUESTED** [review]
Why do we have changes in `ios` folder? Can you try to revert the files under `ios` folder to the `main` branch's state & check if it still works on your end, please?

File: packages/design-system/src/atoms/buttonV2/index.tsx [hunk 10]
```diff
@@ -116,73 +119,84 @@ const BaseButton = ({
...
-                selectable={false}
-                color={stylesBasedOnState.textColor}
-                style={[styles.title]}
-              >
-                {title}
-              </TypographyV2>
-            )}
-          </>
+      <ConditionalWrapper
+        condition={variant === 'icon' && tooltip !== undefined}
```
Comment (teimurjan) [inline packages/design-system/src/atoms/buttonV2/index.tsx:193]:
```suggestion
        condition={variant === 'icon' && typeof tooltip === 'string'}
```

Comment (teimurjan) [reply packages/design-system/src/atoms/buttonV2/index.tsx:193]: It'll be a safer type-check because we only allow strings here.

**Review by teimurjan — APPROVED** [review]
Nice work! :) Address the last comment before merge, please :)

**Review by diegonzs-roll — APPROVED** [review]
